app/Http/Controllers/User/ProfileController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers\User;
+
+use Exception;
+use App\Models\User;
+use Illuminate\Http\Request;
+use App\Traits\FileProcesser;
+use App\Http\Controllers\Controller;
+use App\Repositories\User\UserInterface;
+use App\Http\Requests\UpdateUserRequest;
+use App\Http\Requests\UploadImageRequest;
+
+class ProfileController extends Controller
+{
+    use FileProcesser;
+
+    /** 
+     * @var userRepository
+     * 
+    */
+    protected $userRepository;
+
+    /**
+     * Create a new controller instance.
+     *
+     * @param  UserInterface  $userRepository
+     * @return void
+     */
+    public function __construct(UserInterface $userRepository)
+    {
+        $this->userRepository = $userRepository;
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  int  $id
+     *
+     * @return Illuminate\Http\JsonResponse
+     */
+    public function show($id)
+    {
+        $user = $this->userRepository->find($id);
+
+        return response()->json($user);
+    }
+
+    /**
+     * Update the specified resource in storage.
+     *
+     * @param  App\Http\Requests\UpdateUserRequest  $request
+     * @param  int  $id
+     *
+     * @return Illuminate\Http\JsonResponse
+     */
+    public function update(UpdateUserRequest $request, $id)
+    {
+        try {
+            $user = $this->userRepository->update($id, $request->all());
+
+            return response()->json([
+                'success' => true,
+                'user' => $user,
+            ]);
+        } catch (Exception $e) {
+
+            return response()->json([
+                'success' => false,
+                'message' => $e->getMessage(),
+            ]);
+        }
+    }
+
+    /**
+     * Change user avatar
+     * @param App\Http\Requests\UploadImageRequest $request
```

Comment:
trường hợp này cũng tương tự Docs của function trên ạ
https://github.com/sunh-1294/Training-php-and-vue/pull/39/files#diff-90bf9812d2b7e1d18facf1ded727ae0aR52